core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -179,12 +178,17 @@ public int getPort() {
   }
 
   @Override
-  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
   public List<ServerServiceDefinition> getServices() {
-    List<ServerServiceDefinition> services =
-        new ArrayList<ServerServiceDefinition>(registry.getServices());
-    services.addAll(fallbackRegistry.getServices());
-    return Collections.unmodifiableList(services);
+    if (fallbackRegistry.getServices().isEmpty()) {
```

Comment:
Save the services returned here to avoid calling `fallbackRegistry.getServices()` three times, which isn't free.